vhdbuilder/packer/install-dependencies.sh

```diff
@@ -414,16 +414,27 @@ K8S_VERSIONS="
 1.18.2.1
 "
 for PATCHED_KUBERNETES_VERSION in ${K8S_VERSIONS}; do
-  HYPERKUBE_URL="mcr.microsoft.com/oss/kubernetes/hyperkube:v${PATCHED_KUBERNETES_VERSION}"
-  # NOTE: the KUBERNETES_VERSION will be used to tag the extracted kubelet/kubectl in /usr/local/bin
-  # it should match the KUBERNETES_VERSION format(just version number, e.g. 1.15.7, no prefix v)
-  # in installKubeletAndKubectl() executed by cse, otherwise cse will need to download the kubelet/kubectl again
-  KUBERNETES_VERSION=$(echo ${PATCHED_KUBERNETES_VERSION} | cut -d"_" -f1 | cut -d"-" -f1 | cut -d"." -f1,2,3)
-  # extractHyperkube will extract the kubelet/kubectl binary from the image: ${HYPERKUBE_URL}
-  # and put them to /usr/local/bin/kubelet-${KUBERNETES_VERSION}
-  extractHyperkube "docker"
-  # remove hyperkube here as the one that we really need is pulled later
-  docker image rm $HYPERKUBE_URL
+    if (($(echo ${PATCHED_KUBERNETES_VERSION} | cut -d"." -f2) < 17)); then
+        HYPERKUBE_URL="mcr.microsoft.com/oss/kubernetes/hyperkube:v${PATCHED_KUBERNETES_VERSION}"
+        # NOTE: the KUBERNETES_VERSION will be used to tag the extracted kubelet/kubectl in /usr/local/bin
+        # it should match the KUBERNETES_VERSION format(just version number, e.g. 1.15.7, no prefix v)
+        # in installKubeletAndKubectl() executed by cse, otherwise cse will need to download the kubelet/kubectl again
+        KUBERNETES_VERSION=$(echo ${PATCHED_KUBERNETES_VERSION} | cut -d"_" -f1 | cut -d"-" -f1 | cut -d"." -f1,2,3)
+        # extractHyperkube will extract the kubelet/kubectl binary from the image: ${HYPERKUBE_URL}
+        # and put them to /usr/local/bin/kubelet-${KUBERNETES_VERSION}
+        extractHyperkube "docker"
+        # remove hyperkube here as the one that we really need is pulled later
+        docker image rm $HYPERKUBE_URL
+    else
+        # If the version has a trailing .1 remove it,
+        # acs-mirror.azureedge.net does not have the base image patch .1 nomenclature at the end
+        KUBERNETES_VERSION=${PATCHED_KUBERNETES_VERSION}
+        if (($(echo ${KUBERNETES_VERSION} | tr -d -c "." | wc -m) > 2 && $(echo ${KUBERNETES_VERSION} | rev | cut -d"." -f 1) == 1)); then
+            KUBERNETES_VERSION=$(echo ${KUBERNETES_VERSION} | rev | cut -d"." -f 2- | rev)
+        fi
+        #extract kubectl and kubelet
+        extractKubeBinaries ${KUBERNETES_VERSION}
```

Comment:
So this KUBERNETES_VERSION would be 1.17.3-hotfix.20200601, meaning the extractKubeBinaries  need to extract 1.17.3 out of it for the final file naming /usr/local/bin/kubelet-1.17.3